### PR TITLE
NODE-308 Node did not receive extensions

### PR DESCRIPTION
**Pull request by vsuharnikov**
Issue was appeared once on devnet

### CODE REVIEW

File: src/main/scala/com/wavesplatform/network/RemoteScoreObserver.scala [hunk 1]
```diff
@@ -11,105 +11,96 @@ import scorex.utils.ScorexLogging
...
-        }
+  override def handlerAdded(ctx: ChannelHandlerContext): Unit = {
+    ctx.channel().closeFuture().addListener { channelFuture: ChannelFuture =>
+      val closedChannel = channelFuture.channel()
+      Option(scores.remove(closedChannel)).foreach { removedScore =>
+        log.debug(s"${id(ctx)} Closed, removing score $removedScore")
+      }
+
+      trySwitchToBestIf(ctx, s"Switching to second best channel, because ${id(closedChannel)} was closed") {
+        case Some((currChannel, _)) => currChannel == closedChannel
```
Comment (phearnot) [inline src/main/scala/com/wavesplatform/network/RemoteScoreObserver.scala]:
```scala
case Some((`closedChannel`, _)) => true
```

File: src/main/scala/com/wavesplatform/network/RemoteScoreObserver.scala [hunk 2]
```diff
@@ -11,105 +11,96 @@ import scorex.utils.ScorexLogging
...
-
-  @volatile
-  private var localScore = initialLocalScore
+  private type ScorePair = (Channel, BigInt)
 
   private val scores = new ConcurrentHashMap[Channel, BigInt]
 
-  private def channelWithHighestScore =
+  @volatile private var localScore = initialLocalScore
+  private val currentRequest = new AtomicReference[Option[ScorePair]](None)
```
Comment (phearnot) [inline src/main/scala/com/wavesplatform/network/RemoteScoreObserver.scala]: Pinned channel score doesn't seem to be used anywhere, so why not just have a reference to `Option[Channel]`? That way the code may be simplified even further.

File: src/main/scala/com/wavesplatform/network/RemoteScoreObserver.scala [hunk 3]
```diff
@@ -11,105 +11,96 @@ import scorex.utils.ScorexLogging
...
+
+  private def bestRemotePair: Option[ScorePair] = {
```
Comment (phearnot) [inline src/main/scala/com/wavesplatform/network/RemoteScoreObserver.scala]: Is there really a point in renaming this method?

**Comment by vsuharnikov** [comment]
@phearnot Actually, a score was used in `trySwitchToBestIf`, in `.filter(_ == best)`. So I had to return ScorePair back.

**Pull request merged**